File: go/lfm.go
package main

import (
    "encoding/json"
    "fmt"
    "io/ioutil"
    "math"
    "net/http"
    "os"
    "runtime/debug"
    "strconv"
    "time"
)

func main() {
	debug.SetGCPercent(-1) // disable GC because it's short-living script
    fmt.Println("Fetching data...")
    apiKey := os.Args[1]
    userName := os.Args[2]
    baseUrl := "https://ws.audioscrobbler.com/2.0/?method=user.getinfo&format=json"
    url := fmt.Sprintf("%s&user=%s&api_key=%s", baseUrl, userName, apiKey)
    resp, err := http.Get(url)
    if (err != nil) {
        panic(err)
    }
    defer resp.Body.Close()
    body, err := ioutil.ReadAll(resp.Body)
    if (err != nil) {
        panic(err)
    }
    var dat map[string]interface{}
    if err := json.Unmarshal(body, &dat); err != nil {
        panic(err)
    }
    user := dat["user"].(map[string]interface{})
    // playCount := user["playcount"].(float64)
    playCount, err := strconv.ParseInt(user["playcount"].(string), 10, 32)
    if (err != nil) {
        panic(err)
    }
    registered := user["registered"].(map[string]interface{})
    startUnixTime, err := strconv.ParseInt(registered["unixtime"].(string), 10, 64)
    if err != nil {
        panic(err)
    }
    startDate := time.Unix(startUnixTime, 0)
    endDate := time.Now()
    daysRegistered := endDate.Sub(startDate).Hours() / 24
    avg := float64(playCount) / daysRegistered
    fmt.Println(avg)
    nextAvg := math.Ceil(avg)
    rem := math.Ceil(nextAvg * daysRegistered - float64(playCount))
    fmt.Printf("It's needed to listen to %.f more songs to make avg %v\n", rem, nextAvg)
}

File: js/lfm.js
const https = require('https');

console.log("Fetching data...");
let baseUrl = "https://ws.audioscrobbler.com/2.0/?method=user.getinfo&format=json";
let args = process.argv.slice(2);
let url = `${baseUrl}&user=${args[1]}&api_key=${args[0]}`;
https.get(url, (resp) => {
  let data = '';
  resp.on('data', (chunk) => data += chunk);
  resp.on('end', () => {
    let user = JSON.parse(data).user;
    let playcount = parseInt(user.playcount);
    let registered = parseInt(user.registered.unixtime);
    let now = new Date().getTime() / 1000;
    let daysDiff = (now - registered) / (60 * 60 * 24);
    let avgCount = playcount / daysDiff;
    let nextAvg = Math.ceil(avgCount);
    let nextLeft = Math.ceil(nextAvg * daysDiff - playcount);
    console.log(avgCount);
    console.log(`It's needed to listen to ${nextLeft} more songs to make avg ${nextAvg}`);
  });
}).on("error", (err) => {
  console.log("Error: " + err.message);
});